packages/client-react-streaming/src/ReadableStreamLink.tsx

```diff
@@ -0,0 +1,137 @@
+import { ApolloLink, Observable } from "@apollo/client/index.js";
+import type { FetchResult } from "@apollo/client/index.js";
+
+export type ReadableStreamLinkEvent =
+  | { type: "next"; value: FetchResult }
+  | { type: "completed" }
+  | { type: "error" };
+
+interface InternalContext {
+  [teeToReadableStreamKey]?: ReadableStreamDefaultController<ReadableStreamLinkEvent>;
+  [readFromReadableStreamKey]?: ReadableStream<ReadableStreamLinkEvent>;
+}
+
+const teeToReadableStreamKey = Symbol.for(
+  "apollo.tee.readableStreamController"
+);
+const readFromReadableStreamKey = Symbol.for("apollo.read.readableStream");
+
+/**
+ * Apply to a context that will be passed to a link chain containing `TeeToReadableStreamLink`.
+ * @param controller
+ * @param context
+ * @returns
+ */
+export function teeToReadableStream<T extends Record<string, any>>(
+  controller: ReadableStreamDefaultController<ReadableStreamLinkEvent>,
+  context: T
+): T & InternalContext {
+  return Object.assign(context, {
+    [teeToReadableStreamKey]: controller,
+  });
+}
+
+/**
+ * Apply to a context that will be passed to a link chain containing `ReadFromReadableStreamLink`.
+ * @param readableStream
+ * @param context
+ * @returns
+ */
+export function readFromReadableStream<T extends Record<string, any>>(
+  readableStream: ReadableStream<ReadableStreamLinkEvent>,
+  context: T
+): T & InternalContext {
+  return Object.assign(context, {
+    [readFromReadableStreamKey]: readableStream,
+  });
+}
+
+/**
+ * A link that allows the request to be cloned into a readable stream, e.g. for
+ * transport of multipart responses from RSC or a server loader to the browser.
+ */
+export const TeeToReadableStreamLink = new ApolloLink((operation, forward) => {
```

Comment:
[nit] Since the description calls this "cloning" the request, what about calling the link that?

```suggestion
export const CloneToReadableStreamLink = new ApolloLink((operation, forward) => {
```

Perhaps I'm just overly not fond of "Tee", but feel free to ignore this if you prefer the current name 😆

Reply:
There's a reason for the name in that context :)
https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/tee

Reply:
Oh great! I was looking for that name somewhere, but only read the [`ReadableStream`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/ReadableStream) page and missed this. Makes sense now! Let's leave it.